tools/wptrunner/wptrunner/webdriver_server.py

```diff
@@ -235,24 +225,16 @@ def cmd_arg(name, value=None):
     return rv
 
 
-def get_free_port(start_port, exclude=None):
-    """Get the first port number after start_port (inclusive) that is
-    not currently bound.
-
-    :param start_port: Integer port number at which to start testing.
-    :param exclude: Set of port numbers to skip"""
-    port = start_port
+def get_free_port():
+    """Get a random unbound port"""
     while True:
-        if exclude and port in exclude:
-            port += 1
-            continue
         s = socket.socket()
         try:
-            s.bind(("127.0.0.1", port))
+            s.bind(("127.0.0.1", 0))
```

Comment:
So this kind of introduces a race that wasn't there; if we start lots of processes at once, and have some delay between getting the port and actually using it we could end up reassigning a port we already requested, even assuming that binding to port 0 provides a totally random otherwise unbound port. Now for reasonable numbers of processes the probabilities of a race here are small, something like 0.5% at 25 processes assuming we assign all the ports before we use any of them. And of course there's always the chance that something else on the system grabs a port we want. So I guess this is OK, but it would also be nice to have a mechanism for excluding ports we already plan to use ourselves.

Reply:
Finding a way to share the `_used_ports` set between different processes would solve the initial issue, but from what I've searched this is not as easy as it looks at first.

Reply:
So, I think it's not true that we have similar races all over the place, and you're correct that it's not trivial to solve; maybe the right thing is to accept this patch now and try to do something better in the future.